frontend/public/lib/socialMediaOperations.js

```diff
@@ -0,0 +1,138 @@
+import InstagramIcon from "@material-ui/icons/Instagram";
+import TwitterIcon from "@material-ui/icons/Twitter";
+import YouTubeIcon from "@material-ui/icons/YouTube";
+import FacebookIcon from "@material-ui/icons/Facebook";
+import LinkedInIcon from "@material-ui/icons/LinkedIn";
+
+import { apiRequest } from "./apiOperations";
+
+export function verifySocialMediaLink(socialChannel, url, texts) {
+  let error = "";
+  // matches http://, https:// , https://www. ,http://www.
+  const regexPrefix = "^(http)(?:s)?(://)(?:www.)?";
+  // matches.com/ anything
+  const regexSuffix = ".com/.+$";
+  let regex;
+  let matches = false;
+
+  switch (socialChannel.name) {
+    case "Twitter": // twitter
+      regex = new RegExp(regexPrefix + "twitter" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_twitter;
+      break;
+    case "Youtube": // youtube
+      regex = new RegExp(regexPrefix + "youtube" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_youtube;
+      break;
+    case "LinkedIn": // linkedIn
+      regex = new RegExp(regexPrefix + "linkedin" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_linkedin;
+      break;
+    case "Instagram": // instagram
+      regex = new RegExp(regexPrefix + "instagram" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_instagram;
+      break;
+    case "Facebook": // facebook
+      regex = new RegExp(regexPrefix + "facebook" + regexSuffix);
+      matches = regex.test(url);
+      error = matches ? "" : texts.does_not_comply_facebook;
+      break;
+    default:
+      break;
+  }
+  return error;
+}
+
+export function getSocialMediaButtons(socialLinks) {
+  const socialMediaLinks = [];
+  socialLinks.map((social) => {
+    switch (social.social_media_channel.social_media_name) {
+      case "Twitter": // twitter
+        socialMediaLinks.push(createSocialMediaIconButton(social));
+        break;
+      case "Youtube": // youtube
+        socialMediaLinks.push(createSocialMediaIconButton(social));
+        break;
+      case "LinkedIn": // linkedin
+        socialMediaLinks.push(createSocialMediaIconButton(social));
+        break;
+      case "Instagram": // instagram
+        socialMediaLinks.push(createSocialMediaIconButton(social));
+        break;
+      case "Facebook": // facebook
+        socialMediaLinks.push(createSocialMediaIconButton(social));
+        break;
+      default:
+        break;
```

Comment:
Why do we have 5 cases that all execute the exact same code? :D

Reply:
Refactored, removed unnecessary function call.